roxie/ccd/ccdsnmp.cpp

```diff
@@ -925,9 +854,9 @@ class CQueryStatsAggregator : public CInterface, implements IQueryStatsAggregato
     {
         if (CInterface::Release())
             return true;
-        SpinBlock b(queryStatsCrit);
         if (!IsShared())
         {
+            WriteLockBlock b(queryStatsLock);
```

Comment:
I think this code would be cleaner in beforeDispose() instead.

Reply:
Not sure it would ever get to beforeDispose(). And while I don't disagree the code is messy, I'd rather not make unrelated changes to it that I don't fully understand!

Reply:
I agree it would need extra changes, so not worth doing.